modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java

```diff
@@ -106,7 +110,6 @@ public class MuleArtifactClassLoader extends FineGrainedControlClassLoader imple
   private volatile boolean shouldReleaseIbmMQResources = false;
   private volatile boolean shouldReleaseActiveMQReferences = false;
   private ResourceReleaser jdbcResourceReleaserInstance;
-  private final ResourceReleaser scalaClassValueReleaserInstance;
```

Comment:
moved to local variable in a method explicitly dedicated to the extension-specific resource releasers.

Reply:
I really don't know the whole story behind it. But I've been told that this Releaser was born from an issue with API Kit.

It turns out [API Kit](https://docs.mulesoft.com/release-notes/apikit/apikit-4.1.8.0-release-notes) uses [AMF](https://github.com/aml-org/amf/releases/tag/5.2.2) which in turn uses [Scala](https://github.com/scala/scala/pull/9276).

We had different issues associated to that:
https://www.mulesoft.org/jira/browse/SE-12829
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000010CJRCYA4/view

The first one ended up with the creation of this Releaser. The second one was fixed with a Scala version bump.

I don't even know if the Scala version bump may even render this Releaser completely useless (for newer versions of API Kit).

In any case, it seems we don't really have this leak coming from the Runtime itself or its dependencies. That is why we are considering something connector-related (being API Kit the only known one).

We want to move towards a model in which the one bringing in the leak is the one responsible for releasing those resources.